File: languages/python/bloock/entity/integrity/proof_anchor.py
from __future__ import annotations

from typing import List

import bloock._bridge.proto.bloock_integrity_entities_pb2 as proto
from bloock.entity.integrity.anchor_network import AnchorNetwork


class ProofAnchor:
    """
    Represents a proof anchor.
    """
    def __init__(
            self, anchor_id: int, networks: List[AnchorNetwork], root: str, status: str
    ) -> None:
        """
        Constructs a ProofAnchor object with the specified parameters.
        :type status: object
        :type root: object
        :type networks: object
        :type anchor_id: object
        :rtype: object
        """
        self.anchor_id = anchor_id
        self.networks = networks
        self.root = root
        self.status = status

    @staticmethod
    def from_proto(proof: proto.ProofAnchor) -> ProofAnchor:
        return ProofAnchor(
            anchor_id=proof.anchor_id,
            networks=list(map(lambda x: AnchorNetwork.from_proto(x), proof.networks)),
            root=proof.root,
            status=proof.status,
        )

    def to_proto(self) -> proto.ProofAnchor:
        return proto.ProofAnchor(
            anchor_id=self.anchor_id,
            networks=list(map(lambda x: x.to_proto(), self.networks)),
            root=self.root,
            status=self.status,
        )

File: languages/php/Bloock/Entity/Integrity/ProofAnchor.php
<?php

namespace Bloock\Entity\Integrity;

/**
 * Represents a proof anchor.
 */
class ProofAnchor
{
    private int $anchorId;
    private array $networks;
    private string $root;
    private string $status;

    /**
     * Constructs a ProofAnchor object with the specified parameters.
     * @param int $anchorId
     * @param array $networks
     * @param string $root
     * @param string $status
     */
    public function __construct(int $anchorId, array $networks, string $root, string $status)
    {
        $this->anchorId = $anchorId;
        $this->networks = $networks;
        $this->root = $root;
        $this->status = $status;
    }

    public static function fromProto(\Bloock\ProofAnchor $proofAnchor): ProofAnchor
    {
        $networks = [];
        foreach ($proofAnchor->getNetworks() as $network) {
            $networks[] = AnchorNetwork::fromProto($network);
        }
        return new ProofAnchor($proofAnchor->getAnchorId(), $networks, $proofAnchor->getRoot(), $proofAnchor->getStatus());
    }

    /**
     * Gets the networks of the anchor proof
     * @return array
     */
    public function getNetworks(): array
    {
        return $this->networks;
    }

    /**
     * Gets the anchor id of the anchor proof.
     * @return int
     */
    public function getAnchorId(): int
    {
        return $this->anchorId;
    }

    /**
     * Gets the root of the anchor proof.
     * @return string
     */
    public function getRoot(): string
    {
        return $this->root;
    }

    /**
     * Gets the status of the anchor proof.
     * @return string
     */
    public function getStatus(): string
    {
        return $this->status;
    }

    public function toProto(): \Bloock\ProofAnchor
    {
        $p = new \Bloock\ProofAnchor();
        $p->setAnchorId($this->anchorId);
        $p->setRoot($this->root);
        $p->setStatus($this->status);

        $networks = [];
        foreach ($this->networks as $network) {
            $networks[] = $network->toProto();
        }
        $p->setNetworks($networks);
        return $p;
    }
}